### ISSUE TITLE
Unable to SSH to VM from windows and Ubuntu

### ISSUE THREAD

**Issue by GoogleCodeExporter** [prototipando/google-cloud-sdk#546]
```
Issue running command [gcloud compute ssh].

What steps will reproduce the problem?
For me at least - attempting to ssh to any of my VMs from windows desktop.  I 
have tried removing keys from both local dir and from console. I've even 
reinstalled cloud sdk.

What is the expected output? What do you see instead?
An ssh connection should be formed. Instead, after the keys are created (this 
seems fine - they appear in the meta page of the console, and files are created 
in ~\.ssh) the following is printed:"
Server refused our key
FATAL ERROR: Disconnected: No supported authentication methods available 
(server sent: publickey)
"

Please provide any additional information below.
I have removed and recreated keys, re-installed sdk, and re-authenticated 
(gcloud init, gcloud auth login etc). 

SSH from the project console works OK.

This is true for all my Ubuntu based VMs.  I have even tried to SSH from one VM 
to another - same deal...

Installation information:

Google Cloud SDK [96.0.0]

Platform: [Windows, x86_64]
Python Version: [2.7.10 (default, May 23 2015, 09:40:32) [MSC v.1500 32 bit 
(Intel)]]
Python Location: [C:\python27\python.exe]
Site Packages: [Disabled]

Installation Root: [C:\Users\Peter\AppData\Local\Google\Cloud 
SDK\google-cloud-sdk]
Installed Components:
  core: [2016.02.05]
  bq-win: [2.0.18]
  core-win: [2016.02.05]
  gcloud: []
  windows-ssh-tools: [2016.01.29]
  gsutil: [4.16]
  gsutil-win: [4.16]
  bq: [2.0.18]
System PATH: [C:\Users\Peter\AppData\Local\Google\Cloud 
SDK\google-cloud-sdk\bin\..\bin\sdk;C:\python27\;C:\Users\Peter\AppData\Local\Go
ogle\Cloud SDK\google-cloud-sdk\bin;C:\Program Files 
(x86)\Google\Chrome\Application;c:\Program 
Files\Java\jre7\bin\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\
WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\ATI 
Technologies\ATI.ACE\Core-Static;C:\Python34;C:\Program Files (x86)\MySQL\MySQL 
Fabric 1.5 ]
Cloud SDK on PATH: [True]

Installation Properties: [C:\Users\Peter\AppData\Local\Google\Cloud 
SDK\google-cloud-sdk\properties]
User Config Directory: [C:\Users\Peter\AppData\Roaming\gcloud]
User Properties: [C:\Users\Peter\AppData\Roaming\gcloud\properties]
Current Workspace: [None]
Workspace Config Directory: [None]
[output truncated]

```

Original issue reported on code.google.com by `pe...@alephinsights.com` on 15 Feb 2016 at 10:01

**Comment by GoogleCodeExporter** [comment]
```
related discussions:
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/gce-d
iscussion/uAPNpkBJXi4
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/gce-d
iscussion/1-0nwPVE_PI
```

Original comment by `pe...@alephinsights.com` on 15 Feb 2016 at 10:03

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report. We can reproduce the problem on Windows 10. Did you run 
gcloud compute ssh on Windows 10?
```

Original comment by `gsfow...@google.com` on 18 Feb 2016 at 4:35
- Changed state: **NeedsMoreInfo**

**Comment by GoogleCodeExporter** [comment]
```
Yes, Windows 10 from both admin and normal user CMD.  

But, strangely, I also tried to ssh from another VM - that failed too! 

I have worked around by using puttygen to generate key, add to the project meta 
keys and then connect with putty.
```

Original comment by `pe...@alephinsights.com` on 18 Feb 2016 at 5:01

**Comment by GoogleCodeExporter** [comment]
```
For the other VM failure, what was the VM OS version, command line and error 
message?

Re: the workaround: we now ship an unpatched PuTTY in bin\sdk\putty.exe. Can 
you try running with that PuTTY just to rule out version differences?

thanks
```

Original comment by `gsfow...@google.com` on 18 Feb 2016 at 5:06

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I used my vanilla Ubuntu VM: Ubuntu 14.04.3 LTS
the read out is as follows.

peter@ubutnu:~$ sudo gcloud compute ssh wordpress-1
Warning: Permanently
added '<IP ADDRESS>' (ECDSA) to the list of known hosts.
Permission denied
(publickey).ERROR: (gcloud.compute.ssh) [/usr/bin/ssh] exited with return
code [255]. See https://cloud.google.com/compute/docs/troubleshooting#ssherrors 
for
troubleshooting hints.

Just tried with the PuTTy in C:\Users\[user]\AppData\Local\Google\Cloud
SDK\google-cloud-sdk\bin\sdk. That worked fine using the key I generated
previously.
```

Original comment by `pe...@alephinsights.com` on 18 Feb 2016 at 5:22

**Comment by GoogleCodeExporter** [comment]
```
The problem has been isolated in google-cloud-sdk\bin\sdk\winkeygen.exe. It 
botched  the public key modulus representation in the generated .ppk and .pub 
files. /var/log/secure on the instance had the clue:
  sshd[PID]: error: buffer_get_bignum2_ret: negative numbers not supported
We're doing a few more tests this morning.
```

Original comment by `gsfow...@google.com` on 19 Feb 2016 at 2:27
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
The problem is fixed in Cloud SDK Release 98.0.0. It includes remediation that 
recognizes corrupt Windows key files and regenerates them on the fly.
```

Original comment by `gsfow...@google.com` on 25 Feb 2016 at 1:04
- Changed state: **Fixed**
